lib/features/mailbox/presentation/mailbox_controller.dart

```diff
@@ -550,14 +550,14 @@ class MailboxController extends BaseMailboxController with MailboxActionHandlerM
   ) {
     log('MailboxController::_handleOpenMailbox():MailboxName: ${presentationMailboxSelected.name}');
     KeyboardUtils.hideKeyboard(context);
-    mailboxDashBoardController.setSelectedMailbox(presentationMailboxSelected);
     mailboxDashBoardController.clearSelectedEmail();
     if (presentationMailboxSelected.id != mailboxDashBoardController.selectedMailbox.value?.id) {
       mailboxDashBoardController.clearFilterMessageOption();
     }
     _disableAllSearchEmail();
     _replaceBrowserHistory();
     mailboxDashBoardController.closeMailboxMenuDrawer();
+    mailboxDashBoardController.setSelectedMailbox(presentationMailboxSelected);
```

Comment:
Can you write the test to make sure the order of actions in this case? @tddang-linagora